jaeger-spark-dependencies-cassandra/src/main/java/io/jaegertracing/spark/dependencies/cassandra/CassandraDependenciesJob.java

```diff
@@ -0,0 +1,225 @@
+      sparkProperties.put("spark.cassandra.auth.password", Utils.getEnv("CASSANDRA_PASSWORD", ""));
+    }
+
+    /** When set, this indicates which jars to distribute to the cluster. */
+    public Builder jars(String... jars) {
+      this.jars = jars;
+      return this;
+    }
+
+    /** Keyspace to store dependency rowsToLinks. Defaults to "jaeger_v1_test" */
+    public Builder keyspace(String keyspace) {
+      Utils.checkNoTNull("keyspace", keyspace);
+      this.keyspace = keyspace;
+      return this;
+    }
+
+    /** Day (in epoch milliseconds) to process dependencies for. Defaults to today. */
+    public Builder day(long day) {
+      this.day = Utils.midnightUTC(day);
+      return this;
+    }
+
+    /** Comma separated list of hosts / IPs part of Cassandra cluster. Defaults to localhost */
+    public Builder contactPoints( String contactPoints) {
+      this.contactPoints = contactPoints;
+      return this;
+    }
+
+    /** The local DC to connect to (other nodes will be ignored) */
+    public Builder localDc(String localDc) {
+      this.localDc = localDc;
+      return this;
+    }
+
+    public CassandraDependenciesJob build() {
+      return new CassandraDependenciesJob(this);
+    }
+  }
+
+  private final String keyspace;
+  private final long day;
+  private final String dateStamp;
+  private final SparkConf conf;
+
+  CassandraDependenciesJob(Builder builder) {
+    this.keyspace = builder.keyspace;
+    this.day = builder.day;
+    SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+    df.setTimeZone(TimeZone.getTimeZone("UTC"));
+    this.dateStamp = df.format(new Date(builder.day));
+    this.conf = new SparkConf(true)
+        .setMaster(builder.sparkMaster)
+        .setAppName(getClass().getName());
+    conf.set("spark.cassandra.connection.host", parseHosts(builder.contactPoints));
+    conf.set("spark.cassandra.connection.port", parsePort(builder.contactPoints));
+    if (builder.localDc != null) {
+      conf.set("connection.local_dc", builder.localDc);
+    }
+    if (builder.jars != null) {
+      conf.setJars(builder.jars);
+    }
+    for (Map.Entry<String, String> entry : builder.sparkProperties.entrySet()) {
+      conf.set(entry.getKey(), entry.getValue());
+    }
+  }
+
+  public void run() {
+    long microsLower = day * 1000;
+    long microsUpper = (day * 1000) + TimeUnit.DAYS.toMicros(1) - 1;
+
+    log.info("Running Dependencies job for {}: {} ≤ Span.timestamp {}", dateStamp, microsLower,
+        microsUpper);
+
+    JavaSparkContext sc = new JavaSparkContext(conf);
+    try {
+      JavaPairRDD<String, Iterable<Span>> traces = javaFunctions(sc)
+          .cassandraTable(keyspace, "traces", mapRowTo(Span.class))
+          .where("start_time < ? AND start_time > ?", microsUpper, microsLower)
+          .mapToPair(span -> new Tuple2<>(span.getTraceId(), span))
+          .groupByKey();
+
+      // TODO remove for debug purposes
+      traces.foreach(stringIterableTuple2 -> {
```

Comment:
Ja, once this is approved. Tests aren't still stable :/

Reply:
OK, then put something like "Remove before merging (or after 2017-10-01)", so that you signal to the next maintainer that this can be safely removed if you forget to remove "now".